linalg/operator.hpp

```diff
@@ -462,20 +492,40 @@ class TimeDependentOperator : public Operator
 
    /** @brief Solve for the unknown @a k, at the current time t, the following
        equation:
-       F(@a u + @a gamma @a k, @a k, t) = G(@a u + @a gamma @a k, t).
+        1. $F( \widehat{u} , \dot{u}, t) = G( \widehat{u}, t)$, if solving for stage-state, $k = \widehat{u}$
+        2. $F( u + \gamma  \dot{u}, \dot{u}, t) = G( u +  \gamma \dot{u}, t)$, if solving for stage-slope, $ k = \dot{u} = \frac{du}{dt} $
+
+        where the stage state and slope follow the relation: $\dot{u} = \frac{du}{dt} = \frac{\widehat{u} - u}{\gamma}$.
 
        For solving an ordinary differential equation of the form
        $ M \frac{dy}{dt} = g(y,t) $, recall that F and G can be defined in
-       various ways, e.g.:
-
-         1. F(u,k,t) = k and G(u,t) = inv(M) g(u,t)
-         2. F(u,k,t) = M k and G(u,t) = g(u,t)
-         3. F(u,k,t) = M k - g(u,t) and G(u,t) = 0
-
-       Regardless of the choice of F and G, this function should solve for @a k
-       in M @a k = g(@a u + @a gamma @a k, t).
-
-       To see how @a k can be useful, consider the backward Euler method defined
+       various ways, depending on the stage variable:
+
+        <table>
+        <caption>Various definitions for $~F~$ and $~G~$</caption>
+        <tr>
+          <th>stage-state, $k = \widehat{u}$</th>
+          <th>stage-slope, $k = \dot{u} = \frac{du}{dt}$ </th>
+        </tr>
+        <tr>
+          <td>$F(\widehat{u},\dot{u},t) = \widehat{u}~$ and $~G(u,t) = \gamma M^{-1} g(u,t) + u$</td>
+          <td>$F(u,\dot{u},t) = \dot{u}~$ and $~G(u,t) = M^{-1} g(u,t)$</td>
+        </tr>
+        <tr>
+          <td>$F(\widehat{u},\dot{u},t) = M \widehat{u}~$ and $~G(u,t) = \gamma g(u,t) + M u$</td>
```

Comment:
I'm picking this particular form of $F$ and $G$, but I have a similar concern for the others:

Assuming solving for stage-state, then $F(u,v,t) = M u$ and and $G(u,t) = \gamma g(u,t) + M u$.  Then I believe the comments indicate `ImplicitSolve` will solve for $k$ in $F(k, v, t) = G(k, t)$ which is equivalent to $M k = \gamma g(k,t) + Mk$, i.e., $g(k,t) = 0$... which I don't believe is what is desired for an implicit stage-state solve?  Or am I missing something?

Reply:
Perhaps instead you want something like "solve for $z$ in $F(z,r,t) = G(z)$ where $F(z,r,t) = M z - r$ and $G(z,t) = \gamma g(z,t)$" (I've used `z` and `r` instead of `k` and `v` for my own sanity ... feel free to stick with `k` and `v`)?

The stage-state solve $M z_i = M y^n + \Delta t \sum_{j=0}^{i-1} a_{ij} g(z_j) + \Delta t a_{ii} g(z_i)$ could then be accomplished by passing in $M y^n + \Delta t \sum_{j=0}^{i-1} a_{ij} g(z_j)$ for `r` and solving for $z_i$ in $F(z_i, r, t) = G(z_i)$

Reply:
Fixed in updated documentation